commit aea51ecc6743f22b1cb3f58a5bf676d28a8a35a0
Author: Hou Pengyang <houpengyang@huawei.com>
Date:   Thu Feb 16 12:34:31 2017 +0000

    f2fs: add ovp valid_blocks check for bg gc victim to fg_gc
    
    commit e93b9865251a0503d83fd570e7d5a7c8bc351715 upstream.
    
    For foreground gc, greedy algorithm should be adapted, which makes
    this formula work well:
    
            (2 * (100 / config.overprovision + 1) + 6)
    
    But currently, we fg_gc have a prior to select bg_gc victim segments to gc
    first, these victims are selected by cost-benefit algorithm, we can't guarantee
    such segments have the small valid blocks, which may destroy the f2fs rule, on
    the worstest case, would consume all the free segments.
    
    This patch fix this by add a filter in check_bg_victims, if segment's has # of
    valid blocks over overprovision ratio, skip such segments.
    
    Signed-off-by: Hou Pengyang <houpengyang@huawei.com>
    Reviewed-by: Chao Yu <yuchao0@huawei.com>
    Signed-off-by: Jaegeuk Kim <jaegeuk@kernel.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Index: linuxtest/dummy/rpi_1340_aea51ecc6743f22b1cb3f58a5bf676d28a8a35a0.txt
===================================================================
--- /dev/null
+++ linuxtest/dummy/rpi_1340_aea51ecc6743f22b1cb3f58a5bf676d28a8a35a0.txt
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
